pkg/config/template.go

```diff
@@ -47,3 +70,105 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMManager holds the details needed for the configmap controller
+type TemplateCMManager struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates CiTemplates
+}
+
+// NewTemplateCMManager creates a new TemplateCMManager
+func NewTemplateCMManager(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates CiTemplates) *TemplateCMManager {
+	return &TemplateCMManager{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMManager) CreateCMTemplates() error {
+	var errors []error
+	for filename, template := range c.templates {
+		templateData, err := GetTemplateData(template)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(template.Name, []string{filename, templateData})
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmName,
+				Labels: map[string]string{
+					createByRehearse:  "true",
+					rehearseLabelPull: strconv.Itoa(c.prNumber),
+				},
+			},
+			Data: map[string]string{filename: templateData},
+		}
+
+		c.logger.WithFields(logrus.Fields{"template-name": template.Name, "cm-name": cmName}).Info("creating rehearsal configMap for template")
+		if _, err := c.cmclient.Create(cm); err != nil {
+			if kerrors.IsAlreadyExists(err) {
+				if _, err := c.cmclient.Update(cm); err != nil {
+					errors = append(errors, fmt.Errorf("could not update existing configmap: %v", err))
+				}
+			} else {
+				errors = append(errors, err)
+			}
+		}
+	}
+	return kutilerrors.NewAggregate(errors)
+}
+
+// CleanupCMTemplates deletes all the configMaps that have been created for the changed templates.
+func (c *TemplateCMManager) CleanupCMTemplates() error {
+	c.logger.Info("deleting temporary template configMaps")
+	if err := c.cmclient.DeleteCollection(&metav1.DeleteOptions{},
+		metav1.ListOptions{LabelSelector: fields.Set{
+			createByRehearse:  "true",
+			rehearseLabelPull: strconv.Itoa(c.prNumber),
+		}.AsSelector().String()}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetTempCMName(templateName string, inputs []string) string {
+	return fmt.Sprintf("rehearse-%s-%s", inputHash(inputs), templateName)
```

Comment:
Hm, how the long will this be? How long will be the hash? Won't we hit some limits?

Reply:
The hash is always 5 characters. `rehearse` is 8 and the biggest cm name that we currently have is 45. So in total, we are in 60 characters maximum.